CMakeLists.txt

```diff
@@ -31,7 +31,8 @@ set(VERSION_IS_RELEASE OFF)
 find_program(CCACHE_PROGRAM ccache)
 if(CCACHE_PROGRAM)
     message(STATUS "Found ccache in ${CCACHE_PROGRAM}")
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
+    set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
+    set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
```

Comment:
What about `RULE_LAUNCH_COMPILE` isn't sufficient?

As an aside, none of this is necessary. See: https://github.com/SFML/SFML/pull/2032#issuecomment-1529981268

Reply:
[CMake documentation states that it is intended for internal use.](https://cmake.org/cmake/help/latest/prop_gbl/RULE_LAUNCH_COMPILE.html)

> Note: This property is intended for internal use by ctest(1). Projects and developers should use the `<LANG>_COMPILER_LAUNCHER` target properties or the associated `CMAKE_<LANG>_COMPILER_LAUNCHER` variables instead.

Also, the executable passed to `RULE_LAUNCH_COMPILE` would be used for everything that ends up being "compiled", this includes non-C/C++/C-like files which ccache does not support. This is definitely not the intention.

Also, I don't think anyone is realistically going to set up ccache via environment variable. As a C/C++ developer you are likely going to end up building third party libraries using CMake more often than not. Just setting up an environment variable would result in these libraries also taking space up in the cache even though you only intended to build them once (or at least very infrequently). The advantage ccache provides is speeding up frequent rebuilds, and that is something you only care about for your own projects and not everything that is built on the system.

I am also in favour of removing this block from the configuration and just passing the variables on the command line (or in our special case, even via environment variable since the CI workers are sandboxes). As long as this block is in the configuration the user isn't given a choice of choosing to build SFML with or without ccache if it is installed on their system.

Reply:
> [CMake documentation states that it is intended for internal use.](https://cmake.org/cmake/help/latest/prop_gbl/RULE_LAUNCH_COMPILE.html)

👍🏻

> I am also in favour of removing this block from the configuration and just passing the variables on the command line (or in our special case, even via environment variable since the CI workers are sandboxes). As long as this block is in the configuration the user isn't given a choice of choosing to build SFML with or without ccache if it is installed on their system.

In a later PR we can consider removing this but I'm fine leaving this in place for this PR.